### ISSUE TITLE
SQL-Error -> MySQL strict mode

### ISSUE THREAD

**Issue by Shentao83** [jweiland-net/events2#101]
Hi, 

if I try to save a new Event withouth having Location or Organizer set, I get an error message. 

> SQL error: 'SQLSTATE[HY000]: General error: 1366 Incorrect integer value: '' for column 'location' at row 1' (tx_events2_domain_model_event:NEW5c73c1ca03c31728674411)

or 

> SQL error: 'SQLSTATE[HY000]: General error: 1366 Incorrect integer value: '' for column 'organizer' at row 1' (tx_events2_domain_model_event:NEW5c73c1eb086bb974155914)

As far as I have seen on another thread, the solution would be to define the default value string 

`'default' => 0`

See https://stackoverflow.com/questions/40542151/typo3-sql-error-incorrect-integer-value-for-column-sys-language-uid-at

**Comment by Shentao83** [comment]
If you add 

`'default' => 0,`

to the file /typo3conf/ext/events2/Configuration/TCA/tx_events2_domain_model_event.php

to the columns location|config and organizer|config, everything works fine so far.

**Comment by Shentao83** [comment]
```
        'location' => [
            'exclude' => true,
            'label' => 'LLL:EXT:events2/Resources/Private/Language/locallang_db.xlf:tx_events2_domain_model_event.location',
            'config' => [
                'type' => 'group',
                'internal_type' => 'db',
                'allowed' => 'tx_events2_domain_model_location',
                'foreign_table' => 'tx_events2_domain_model_location',
                'size' => 1,
                'maxitems' => 1,
                'minitems' => 0,
                'suggestOptions' => [
                    'default' => [
                        'searchWholePhrase' => true,
                    ],
                ],
                'default' => 0,
            ],
        ],
        'organizer' => [
            'exclude' => true,
            'label' => 'LLL:EXT:events2/Resources/Private/Language/locallang_db.xlf:tx_events2_domain_model_event.organizer',
            'config' => [
                'type' => 'group',
                'internal_type' => 'db',
                'allowed' => 'tx_events2_domain_model_organizer',
                'foreign_table' => 'tx_events2_domain_model_organizer',
                'size' => 1,
                'maxitems' => 1,
                'minitems' => 0,
                'suggestOptions' => [
                    'default' => [
                        'searchWholePhrase' => true,
                    ],
                ],
                'default' => 0,
            ],
        ],
```

**Comment by froemken** [comment]
Good Morning @Shentao83 

sorry for my late answer. I have updated events2 18 days ago to solve that problem:
https://github.com/jweiland-net/events2/commit/f86843be288ddeb2e007b749830fdc5ba84805fe

This problem is solved with version 3.4.1.
Thank you

Stefan
